src/native/s3select/s3select_napi.cpp

```diff
@@ -0,0 +1,251 @@
+/* Copyright (C) 2016 NooBaa */
+
+#include "../../../src/native/third_party/isa-l/include/crc.h"
+#include "../../../submodules/s3select/include/s3select.h"
+#include "../util/common.h"
+#include "../util/napi.h"
+#include <arpa/inet.h>
+
+namespace noobaa
+{
+
+enum InputFormat
+{
+    csv,
+    json,
+    parquet
+};
+
+class SelectWorker : public Napi::AsyncWorker
+{
+public:
+    Napi::Promise::Deferred deferred;
+    s3selectEngine::csv_object& csv_object;
+    s3selectEngine::json_object& json_object;
+    std::string select;
+    const char* buffer;
+    const unsigned int buffer_len;
+    const bool is_flush;
+    const InputFormat input_format;
+    const u_int8_t* headers_bytes;
+    const u_int32_t headers_len;
```

Comment:
I suggest that you rename all members to start with `_` - this is the best way to identify what is a class member vs what is a local variable when reading the code flow.

Reply:
@guymguym, a prefix of `_` in a member name is usually a convention of a private member, from what I know `m` or `m_` prefix is for public members (It is a separate discussion about why those members are public, and I will not start it at this point...).

Reply:
You're right, but these are all "effectively" private members, the fact their visibility is public is unrelated to their use in the code and we never really share these classes outside this module. 

To be honest I am not a fan of encapsulation (although I was once a fan of it) but the more I used javascript/python and other dynamic languages, I got used to the model that anything is effectively public, and it is the code's job to explain why something shouldn't be used directly or how it should be used. 

In any case, for c++ classes/structs (which are not plain data objects but actually types with methods), the reason for prefixing members with `_` is just to make it easier to read their methods and quickly identify that we are accessing a member and not a local variable or function argument, because in c++ there is no requirement to use `this->` when accessing members.

Reply:
Oh and I would refrain from adding `m_` mainly because this changes the pronunciation of the names and makes it harder to express logic with it. e.g. `open(m_file_name, m_mode, m_options)` - yuck :)